pandas/core/arrays/integer.py

```diff
@@ -103,6 +103,10 @@ def __from_arrow__(
         import pyarrow  # noqa: F811
         from pandas.core.arrays._arrow_utils import pyarrow_array_to_numpy_and_mask
 
+        pyarrow_type = pyarrow.from_numpy_dtype(self.type)
+        if not array.type.equals(pyarrow_type):
+            array = array.cast(pyarrow_type)
```

Comment:
this is the cause of a bunch of xfails in test_from_arrow_type_error.  does this need to be tightened somehow, or is the test behavior not something we care about?

Reply:
It _could_ be tightened, for example by checking that the `array.type` is integral